allocate.go

```diff
@@ -105,11 +106,15 @@ type ExecAllocator struct {
 	initFlags map[string]interface{}
 	initEnv   []string
 
-	wg sync.WaitGroup
+	modifyCmdFunc func(cmd *exec.Cmd)
 
-	combinedOutputWriter io.Writer
+	// Chrome will sometimes fail to print the websocket, or run for a long
+	// time, without properly exiting. To avoid blocking forever in those
+	// cases, give up after a specified timeout.
+	wsURLReadTimeout time.Duration
 
-	modifyCmdFunc func(cmd *exec.Cmd)
+	wg                   sync.WaitGroup
```

Comment:
Do not remove the blank line after `wg sync.WaitGroup` because `wg` is not totally for `combinedOutputWriter`.

Reply:
reference:

https://github.com/chromedp/chromedp/blob/621263b43cbf8797c256ffac2508d955f3f8ad95/allocate.go#L206-L209

Reply:
II guess I  just do not understand your request 1. sorry :( 

You said wg works like a "mutex hat". Reading about mutex hats, from your link - "It sits, like a hat, on top of the variables that it protects." 

What does it protect here?

I just don't understand what is the purpose of moving `modifyCmdFunc func(cmd *exec.Cmd)` up. And if it should be a separate commit, what should be reason for that in that commit.

Reply:
Sorry, it's my fault that I haven't made it clear. First of all, this is a personal taste. So it's a little hard to understand. I will try to explain it.

First, see the source code below:

https://github.com/chromedp/chromedp/blob/621263b43cbf8797c256ffac2508d955f3f8ad95/allocate.go#L206-L225

https://github.com/chromedp/chromedp/blob/621263b43cbf8797c256ffac2508d955f3f8ad95/allocate.go#L234-L237

https://github.com/chromedp/chromedp/blob/621263b43cbf8797c256ffac2508d955f3f8ad95/allocate.go#L244-L248

As the comments in line 206 and 208 said, `wg` is for both the entire allocator and the io.Copy in a separate goroutine. That's why I said `wg` **works like a mutex hat**. Strictly speaking, It does not protect `combinedOutputWriter`, but it has something to do with `combinedOutputWriter`. So I think they should be arranged together in the struct.

On the other hand, `wg` is not totally for `combinedOutputWriter` (it's for the entire allocator too). If we remove the blank line between them, it will make it looks like that `wg` is totally for `combinedOutputWriter`.

So I think the best choice is to put `wg` and `combinedOutputWriter` into the last part of the struct and leave a blank line between them. This is what it's arranged before `modifyCmdFunc func(cmd *exec.Cmd)` is added in #674.

This suggested change has nothing to do with `wsURLReadTimeout`, and it's hard to explain. That's why it deserves a standalone commit.

Reply:
@ZekeLu If this change has nothing to do with `wsURLReadTimeout`, then why it is even discussed in this PR?

Reply:
@8tomat8 
```
+// WSURLReadTimeout sets the waiting time for reading the WebSocket URL.
+// The default value is 20 seconds.
+func WSURLReadTimeout(t time.Duration) ExecAllocatorOption {
+   return func(a *ExecAllocator) {
+       a.wsURLReadTimeout = t
+   }
+}
+
```

Reply:
> @ZekeLu If this change has nothing to do with `wsURLReadTimeout`, then why it is even discussed in this PR?

I have submitted 1c2b99f so we can ignore this part now.